syntax = "proto3";

package raft;

message Request_Vote {
	int32 term = 1;
	string candidateId = 2;
	int32 lastLogIndex = 3;
	int32 lastLogTerm = 4;
}

message RequestVoteResponse {
	int32 term = 1;
	bool voteGranted = 2;
}

message Entry {
	string key = 1;
	string value = 2;
}

message Append_Entries {
	int32 term = 1;
	string leaderId = 2;
	int32 prevLogIndex = 3;
	int32 prevLogTerm = 4;
	repeated Entry entries = 5;
	int32 leaderCommit = 6;
}

message AppendEntriesResponse {
	int32 term = 1;
	bool success = 2;
}

service Raft {
	rpc RequestVote(Request_Vote) returns (RequestVoteResponse);
	rpc AppendEntries(Append_Entries) returns (AppendEntriesResponse);
}
